### PR TITLE
fix: ECDSA and ED25519 public key mismatch when you get it from mnemonic

### PR DESCRIPTION
**Pull request by ivaylonikolov7**
**Description**: 

This PR modifies the crypto package and in particular the ECDSA and ED25519 private key classes. A user found out that a mnemonic and a key that share the same private key mismatch their public key. In order to fix it, this PR modifies the ECDSA and ED25519 so they don't add trailing zeroes. This addition only happens only when the private key is shorter than 32 bytes. Example: 3030020100300706052b8104000a04220420a5b9dedc7c7767c4a88a0ac3cc6b9171948bada579226bd550d3342db11dc800

The last 2 trailing zeroes should be added after the DER prefix  bytes (3030020100300706052b8104000a04220420) 
e.g 3030020100300706052b8104000a0422042000a5b9dedc7c7767c4a88a0ac3cc6b9171948bada579226bd550d3342db11dc8

**Fixes**

 #2419

**Notes for reviewer**:
Not sure if 100 tests are enough of a stress test but otherwise the test gets slow. For example 10000 iterations take around 15 seconds per test case. Considering there is 4 of these per file, thats gonna take 2 minute in total which in my opiion is a lot. 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/unit/Ed25519PrivateKey.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ const DER_PRIVATE_KEY_BYTES = new Uint8Array([
 const DER_PUBLIC_KEY =
     "302a300506032b65700321004a6892f034d2d1c9b1a76acca8e34884055172f4210a0c02e3c7d55084f224d1";
 
+const STRESS_TEST_ITERATION_COUNT = 100;
```
Comment (gsstoykov) [inline test/unit/Ed25519PrivateKey.js:18]: Try a bigger value here. There are no heavy calls. In theory you could even try more than 10^7.

Comment (ivaylonikolov7) [reply test/unit/Ed25519PrivateKey.js:18]: It takes a lot of time :( 10 000 tests take 50 seconds to run. Therefore 1000 test take like 5 seconds. 

Now that I think about it... considering the CI takes 30 minutes to run, then 1 minute per test case isn't that much. 

But they are 8 tests in total, so basically 8 minutes addition.

Comment (svetoslav-nikol0v) [reply test/unit/Ed25519PrivateKey.js:18]: 100 is too small and doesn't guarantee that the method works correctly.
Increase tests to 1000 and timeout to 200000.

**Review by svetoslav-nikol0v — APPROVED** [review]

**Review by agadzhalov — APPROVED** [review]
LGTM

**Pull request merged**